{"name": "测试数据：【 测试test，123456，False，1003，用户名不存在】", "status": "broken", "statusDetails": {"message": "requests.exceptions.MissingSchema: Invalid URL '/login': No schema supplied. Perhaps you meant http:///login?", "trace": "self = <testcases.api_test.test_03_login.TestUserLogin object at 0x000001667D46EE50>\nusername = '测试test', password = '123456', except_result = False\nexcept_code = 1003, except_msg = '用户名不存在'\n\n    @allure.story(\"用例--登录用户\")\n    @allure.description(\"该用例是针对获取用户登录接口的测试\")\n    @allure.issue(\"https://www.cnblogs.com/wintest\", name=\"点击，跳转到对应BUG的链接地址\")\n    @allure.testcase(\"https://www.cnblogs.com/wintest\", name=\"点击，跳转到对应用例的链接地址\")\n    @allure.title(\"测试数据：【 {username}，{password}，{except_result}，{except_code}，{except_msg}】\")\n    @pytest.mark.single\n    @pytest.mark.parametrize(\"username, password, except_result, except_code, except_msg\",\n                             api_data[\"test_login_user\"])\n    def test_login_user(self, username, password, except_result, except_code, except_msg):\n        logger.info(\"*************** 开始执行用例 ***************\")\n>       result = login_user(username, password)\n\ntest_03_login.py:28: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\..\\operation\\user.py:90: in login_user\n    res = user.login(data=payload, headers=header)\n..\\..\\api\\user.py:25: in login\n    return self.post(\"/login\", **kwargs)\n..\\..\\core\\rest_client.py:16: in post\n    return self.request(url, \"POST\", data, json, **kwargs)\n..\\..\\core\\rest_client.py:37: in request\n    return requests.post(url, data, json, **kwargs)\n..\\..\\venv\\lib\\site-packages\\requests\\api.py:116: in post\n    return request('post', url, data=data, json=json, **kwargs)\n..\\..\\venv\\lib\\site-packages\\requests\\api.py:60: in request\n    return session.request(method=method, url=url, **kwargs)\n..\\..\\venv\\lib\\site-packages\\requests\\sessions.py:510: in request\n    prep = self.prepare_request(req)\n..\\..\\venv\\lib\\site-packages\\requests\\sessions.py:443: in prepare_request\n    p.prepare(\n..\\..\\venv\\lib\\site-packages\\requests\\models.py:313: in prepare\n    self.prepare_url(url, params)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <PreparedRequest [POST]>, url = '/login', params = OrderedDict()\n\n    def prepare_url(self, url, params):\n        \"\"\"Prepares the given HTTP URL.\"\"\"\n        #: Accept objects that have string representations.\n        #: We're unable to blindly call unicode/str functions\n        #: as this will include the bytestring indicator (b'')\n        #: on python 3.x.\n        #: https://github.com/requests/requests/pull/2238\n        if isinstance(url, bytes):\n            url = url.decode('utf8')\n        else:\n            url = unicode(url) if is_py2 else str(url)\n    \n        # Remove leading whitespaces from url\n        url = url.lstrip()\n    \n        # Don't do any URL preparation for non-HTTP schemes like `mailto`,\n        # `data` etc to work around exceptions from `url_parse`, which\n        # handles RFC 3986 only.\n        if ':' in url and not url.lower().startswith('http'):\n            self.url = url\n            return\n    \n        # Support for unicode domain names and paths.\n        try:\n            scheme, auth, host, port, path, query, fragment = parse_url(url)\n        except LocationParseError as e:\n            raise InvalidURL(*e.args)\n    \n        if not scheme:\n            error = (\"Invalid URL {0!r}: No schema supplied. Perhaps you meant http://{0}?\")\n            error = error.format(to_native_string(url, 'utf8'))\n    \n>           raise MissingSchema(error)\nE           requests.exceptions.MissingSchema: Invalid URL '/login': No schema supplied. Perhaps you meant http:///login?\n\n..\\..\\venv\\lib\\site-packages\\requests\\models.py:387: MissingSchema"}, "description": "该用例是针对获取用户登录接口的测试", "attachments": [{"name": "log", "source": "d145e9b5-ba2f-4a46-9991-c635a94090bb-attachment.txt", "type": "text/plain"}, {"name": "stderr", "source": "d1b5aa1d-ce0a-4d54-a364-1af60a4c97a9-attachment.txt", "type": "text/plain"}], "parameters": [{"name": "username", "value": "'测试test'"}, {"name": "password", "value": "'123456'"}, {"name": "except_result", "value": "False"}, {"name": "except_code", "value": "1003"}, {"name": "except_msg", "value": "'用户名不存在'"}], "start": 1623304905718, "stop": 1623304905720, "uuid": "17b7f37a-dcc0-4e7a-a7ac-c20ee28b6c9e", "historyId": "56bfc4b39b0fa959e074da77d0878485", "testCaseId": "7272db244dabef390096479ae3196642", "fullName": "testcases.api_test.test_03_login.TestUserLogin#test_login_user", "labels": [{"name": "severity", "value": "normal"}, {"name": "feature", "value": "用户登录模块"}, {"name": "epic", "value": "针对单个接口的测试"}, {"name": "story", "value": "用例--登录用户"}, {"name": "tag", "value": "single"}, {"name": "parentSuite", "value": "testcases.api_test"}, {"name": "suite", "value": "test_03_login"}, {"name": "subSuite", "value": "TestUserLogin"}, {"name": "host", "value": "DESKTOP-FRVNKTB"}, {"name": "thread", "value": "21304-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "testcases.api_test.test_03_login"}], "links": [{"type": "test_case", "url": "https://www.cnblogs.com/wintest", "name": "点击，跳转到对应用例的链接地址"}, {"type": "issue", "url": "https://www.cnblogs.com/wintest", "name": "点击，跳转到对应BUG的链接地址"}]}